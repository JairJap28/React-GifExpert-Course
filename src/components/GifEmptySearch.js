import React from 'react';

const EmptySearch = () => {
    return (
        <div>
            <p>There is no data to show yet, please enter a valid keyword.</p>
        </div>
    );
}

export default EmptySearch;