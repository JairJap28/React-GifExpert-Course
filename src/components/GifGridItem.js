import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url }) => {
    return (
        <div className="card animate__animated animate__fadeIn">
            <img className="img__normal" src={url} alt={title}/>
            <img className="img__blur" src={url} alt={`blur ${title}`}/>
            <p>{title || 'No title description'}</p>
        </div>
    )
};

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}