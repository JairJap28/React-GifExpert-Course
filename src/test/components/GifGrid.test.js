import React from 'react';
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Test on GifGrid component', () => {
    const category = 'Time';
    
    test('Component must match with the snapshot', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('Should show items when images are loaded with useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/anything.jpg',
            title: 'Anything'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
});