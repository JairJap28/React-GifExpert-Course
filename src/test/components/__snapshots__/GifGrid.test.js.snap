// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifGrid component Component must match with the snapshot 1`] = `
<Fragment>
  <h3
    className="title__category animate__animated animate__fadeIn"
  >
    Time
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
